#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material2 "slippy"
      coulombFriction [
        0
      ]
    }
    ContactProperties {
      material2 "block"
      coulombFriction [
        0.01
      ]
    }
    ContactProperties {
      material2 "body"
      coulombFriction [
        0
      ]
    }
    ContactProperties {
      material2 "wheel"
      coulombFriction [
        0.8
      ]
    }
  ]
}
Viewpoint {
  orientation -0.34583266624878084 -0.915865279846214 -0.20393762803238505 1.1482986477303605
  position -3.3449255633869317 1.882206041069844 1.6809644738054992
}
Background {
  skyColor [
    0.15 0.45 1
  ]
}
Floor {
  size 2.4 2.4
  appearance Asphalt {
  }
}
DEF Floor_colour Solid {
  translation 0 0.0001 0.4
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Plane {
        size 0.4 0.4
      }
    }
  ]
  name "solid(6)"
}
DEF Floor_colour Solid {
  translation 0 0.0001 -0.4
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
      }
      geometry Plane {
        size 0.4 0.4
      }
    }
  ]
  name "solid(5)"
}
DEF WALL_1 Solid {
  translation 1.2 0.05 0
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0 0.333333 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 2.38
      }
    }
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_2 Solid {
  translation -1.2 0.05 0
  children [
    USE WALL_SHAPE
  ]
  name "solid(1)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation 0 0.05 -1.2
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(2)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation 0 0.05 1.2
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(3)"
  boundingObject USE WALL_SHAPE
}
TexturedBackgroundLight {
  texture "empty_office"
}
target {
  translation 0.7 0.03 0
  colour 0 1 0
  name "target(2)"
}
Robot {
  translation -0.3 0.06 -0.77
  children [
    Receiver {
      type "infra-red"
      channel 2
    }
    Emitter {
      type "infra-red"
      channel 1
    }
    Compass {
      yAxis FALSE
    }
    GPS {
      rotation 0 1 0 -1.5707953071795862
    }
    DEF BODY Group {
      children [
        Transform {
          translation 0 0 0.09
          children [
            Shape {
              appearance DEF TEXTURE PBRAppearance {
                baseColor 1 0 0
                roughness 0.5
                metalness 0.5
              }
              geometry Box {
                size 0.22 0.08 0.006
              }
            }
          ]
        }
        Transform {
          translation 0 0 -0.09
          children [
            Shape {
              appearance USE TEXTURE
              geometry Box {
                size 0.22 0.08 0.006
              }
            }
          ]
        }
        Transform {
          translation -0.11 0 0
          rotation 0 1 0 1.5707996938995747
          children [
            Shape {
              appearance USE TEXTURE
              geometry Box {
                size 0.18 0.08 0.006
              }
            }
          ]
        }
        Transform {
          translation 0.03 0 0
          rotation 0 1 0 1.5707996938995747
          children [
            Shape {
              appearance USE TEXTURE
              geometry Box {
                size 0.18 0.08 0.006
              }
            }
          ]
        }
        Transform {
          translation -0.04 -0.037 0
          rotation 1 0 0 1.5707996938995747
          children [
            Shape {
              appearance USE TEXTURE
              geometry Box {
                size 0.14 0.18 0.006
              }
            }
          ]
        }
        Transform {
          translation 0 0.037 0
          rotation 1 0 0 1.5707996938995747
          children [
            Shape {
              appearance USE TEXTURE
              geometry Box {
                size 0.22 0.18 0.006
              }
            }
          ]
        }
        Transform {
          translation -0.1 -0.05 0
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Sphere {
                radius 0.01
                subdivision 2
              }
            }
          ]
        }
        Transform {
          translation -0.093 0.046 0
          children [
            Shape {
              appearance USE TEXTURE
              geometry Box {
                size 0.04 0.012 0.02
              }
            }
          ]
        }
        Transform {
          translation -0.133 0.055 0
          children [
            Shape {
              appearance USE TEXTURE
              geometry Box {
                size 0.12 0.006 0.02
              }
            }
          ]
        }
        Transform {
          translation -0.18675 0.049 0
          children [
            Shape {
              appearance USE TEXTURE
              geometry Box {
                size 0.0125 0.006 0.02
              }
            }
          ]
        }
      ]
    }
    LightSensor {
      translation -0.18675 0.039 0
      rotation 0 0 1 -1.5707996938995747
      name "ls_red"
      lookupTable [
        0 0 0
        0.39 0 0
        0.4 5 0
      ]
      colorFilter 1 0 0
    }
    LightSensor {
      translation -0.18675 0.039 0
      rotation 0 0 1 -1.5707996938995747
      name "ls_green"
      lookupTable [
        0 0 0
        0.39 0 0
        0.4 5 0
      ]
      colorFilter 0 1 0
    }
    DistanceSensor {
      translation -0.115 -0.03 0
      rotation 0 1 0 3.14
      name "ds_long"
      model "SHARP long distance sensor 20-150cm"
      lookupTable [
        0.15 2.75 0.015
        0.2 2.51 0.015
        0.3 1.99 0.015
        0.4 1.52 0.015
        0.5 1.25 0.015
        0.6 1.04 0.015
        0.7 0.87 0.015
        0.8 0.79 0.015
        0.9 0.74 0.015
        1 0.69 0.015
        1.1 0.6 0.015
        1.2 0.55 0.015
        1.3 0.5 0.015
        1.4 0.47 0.015
        1.5 0.45 0.015
      ]
    }
    DistanceSensor {
      translation -0.115 0 0
      rotation 0 1 0 3.1415
      name "ds_short"
      model "SHARP short distance sensor 10-80cm"
      lookupTable [
        0.05 3.1 0.015
        0.1 2.26 0.015
        0.2 1.27 0.015
        0.3 0.92 0.015
        0.4 0.75 0.015
        0.5 0.6 0.015
        0.6 0.5 0.015
        0.7 0.45 0.015
        0.8 0.41 0.015
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor 0.01 -0.01 0
      }
      device [
        RotationalMotor {
          name "right_wheel"
          maxTorque 0.4
        }
      ]
      endPoint Solid {
        translation 0.009996206181304118 -0.010003329099082333 0.10999995374205494
        rotation -0.782528954456969 -0.4402554365692715 -0.4402540016941166 4.469582737613674
        children [
          DEF WHEEL Shape {
            appearance DEF WHEEL_TEXTTURE PBRAppearance {
            }
            geometry Cylinder {
              height 0.02
              radius 0.05
            }
          }
        ]
        contactMaterial "wheel"
        boundingObject USE WHEEL
        physics Physics {
          density -1
          mass 0.1
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor 0.01 -0.01 0
      }
      device [
        RotationalMotor {
          name "left_wheel"
          maxTorque 0.4
        }
      ]
      endPoint Solid {
        translation 0.009996205168044834 -0.010003331694560861 -0.10999995313745702
        rotation -0.7825286893095337 -0.44025585445254484 -0.440254055095195 4.469581941280483
        children [
          USE WHEEL
        ]
        name "solid(1)"
        contactMaterial "wheel"
        boundingObject USE WHEEL
        physics Physics {
          density -1
          mass 0.1
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor 0.112 -0.005 0.086
      }
      device [
        RotationalMotor {
          name "right_door"
          maxPosition 1.5707
          maxTorque 0.35
        }
      ]
      endPoint Solid {
        translation 0.1301179234595816 -0.0050044816746603465 0.045000003547909076
        rotation 1.3283486500840945e-07 0.9999999999999682 -2.1402072137582648e-07 1.221699787531319
        children [
          DEF DOOR Shape {
            appearance PBRAppearance {
            }
            geometry Box {
              size 0.09 0.07 0.006
            }
          }
        ]
        name "solid(2)"
        boundingObject USE DOOR
        physics Physics {
          density 700
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.112 -0.005 -0.086
      }
      device [
        RotationalMotor {
          name "left_door"
          maxPosition 1.57
          maxTorque 0.35
        }
      ]
      endPoint Solid {
        translation 0.13011792343596756 -0.005004484308355102 -0.04500000415006315
        rotation -1.328623123667678e-07 -0.9999999999999682 -2.1400088446675564e-07 1.2216997875345956
        children [
          USE DOOR
        ]
        name "solid(3)"
        boundingObject USE DOOR
        physics Physics {
          density 700
        }
      }
    }
    Solid {
      translation -0.1 0 0
      children [
        DEF BATTERY Shape {
          appearance USE TEXTURE
          geometry Box {
            size 0.02 0.073 0.06
          }
        }
      ]
      name "battery"
      boundingObject USE BATTERY
      physics Physics {
        density -1
        mass 0.2
      }
    }
    Solid {
      translation 0.0075 0 -0.045
      children [
        DEF MOTOR Shape {
          appearance USE TEXTURE
          geometry Box {
            size 0.04 0.04 0.084
          }
        }
      ]
      name "motor_left"
      boundingObject USE MOTOR
      physics Physics {
        density -1
        mass 0.19
      }
    }
    Solid {
      translation 0.0075 0 0.045
      children [
        DEF MOTOR Shape {
          appearance USE TEXTURE
          geometry Box {
            size 0.04 0.04 0.084
          }
        }
      ]
      name "motor_left(1)"
      boundingObject USE MOTOR
      physics Physics {
        density -1
        mass 0.19
      }
    }
  ]
  contactMaterial "body"
  boundingObject USE BODY
  physics Physics {
    density 700
  }
  controller "endzone_avoidance"
}
